Engine/ac/gui.cpp

```diff
@@ -660,7 +657,11 @@ void unexport_all_gui_controls()
     {
         unexport_gui_controls(i);
     }
-    StaticGUIControlsArray.clear();
+    StaticGUIControlsHandles.clear();
+
+    // FIX-ME: for some reason this method is called in DoBeforeRestore
+    // so we must reset arrays sizes below. Remove if removed from DoBeforeRestore
+    set_array_all_gui_controls_size();
```

Comment:
Why is it necessary to recreate and resize this array after unregistering control exports?

Reply:
There's a fix-me already there in `DoBeforeRestore`. I can't figure why it's called there, I wonder if this is something related to GUI Controls being different between saves - like, because the game was edited and then it's loading an old save on a newer game, but I couldn't figure how that would fix things.

Reply:
I know, but I was wondering why do you clear the array and then immediately recreate it. That code did not make any sense to me.

Reply:
The issue is because it's an array of vector arrays the memory access when restoring when the "sub-array" is not created it will instantaneously crash. But even if it didn't, because the arrays is cleared it's size would not be correct - if the size dynamically adjusted it would be then cause the issue of the memory pointed inside the array being incorrect.

Reply:
i know that of course; my question was, why clear it in the first place, if you are going to recreate it in the very next line?